### PR TITLE
Test simple parquet scheme

### PR DESCRIPTION
**Pull request by mrocklin**
Took a quick stab at supporting single-file parquet datasets from dask.dataframe.  Added a test here.  Also tried making the following changes to fastparquet

```diff
--- a/fastparquet/api.py
+++ b/fastparquet/api.py
@@ -45,14 +45,16 @@ class ParquetFile(object):
         try:
             fn2 = sep.join([fn, '_metadata'])
             f = open_with(fn2, 'rb')
+            with f as f:
+                self._parse_header(f, verify)
             fn = fn2
         except (IOError, OSError):
             f = open_with(fn, 'rb')
+            with f as f:
+                self._parse_header(f, verify)
         self.open = open_with
         self.fn = fn
         self.sep = sep
-        with f as f:
-            self._parse_header(f, verify)
         self._read_partitions()
 
     def _parse_header(self, f, verify=True):
@@ -106,8 +108,11 @@ class ParquetFile(object):
         self.cats = {key: list(v) for key, v in cats.items()}
 
     def row_group_filename(self, rg):
-        return self.sep.join([os.path.dirname(self.fn),
-                              rg.columns[0].file_path])
+        if rg.columns[0].file_path:
+            return self.sep.join([os.path.dirname(self.fn),
+                                  rg.columns[0].file_path])
+        else:
+            return self.fn
```

Though now I have to step out for the day.  cc @martindurant

### CODE REVIEW

File: dask/dataframe/io/tests/test_parquet.py [hunk 1]
```diff
@@ -111,3 +111,12 @@ def test_optimize(fn, c):
...
     assert len(dsk) == x.npartitions
     assert all(v[4] == c for v in dsk.values())
+
+
+def test_roundtrip_from_pandas():
+    with tmpfile() as fn:
+        df = pd.DataFrame({'x': [1, 2, 3]})
+        fastparquet.write(fn, df)
+        ddf = dd.io.parquet.read_parquet(fn)
+        import pdb; pdb.set_trace()
```
Comment (martindurant) [inline dask/dataframe/io/tests/test_parquet.py]: This is a leftover?

**Comment by martindurant** [comment]
I guess this will then be a PR in fastparquet. May as well have `with open_with(fn, 'rb')` in one go. Otherwise, I see no problem - if loading from a single file within fastparquet, `row_group_filename` is never called.

To be sure: I don't think that to_parquet should support single-file mode on write, as that would only be possible for a one-division dataframe. I was, however, thinking of the possibility of a simple function that could collect several isolated parquet files into a logical collection, if they had compatible schemas, of course.

**Comment by martindurant** [comment]
Were you planning on merging this here, and the changes above into fastparquet?

**Comment by mrocklin** [comment]
Honestly I had forgotten about it.  I've been swamped with a few other things.  I'll try to get back to it in a couple of days.  Feel free to steal it from me if you have time.

**Comment by martindurant** [comment]
NB failure fixed in https://github.com/dask/fastparquet/pull/34/ - is this test skipped in appveyor?

**Comment by martindurant** [comment]
This now passes following merger in fastparquet.

**Comment by mrocklin** [comment]
Should we merge?

**Comment by martindurant** [comment]
I believe yes - adding a test that passes has to be a good thing!

**Pull request merged**